### ISSUE TITLE
console.log strangeness in handler function

### ISSUE THREAD

**Issue by drmrbrewer** [nfriedly/express-rate-limit#310]
### Description

I suspect that this isn't a bug in the library, but some other javascript or node.j weirdness I haven't come across, but I thought I'd report it here in case anyone had any ideas...

I'm defining a custom handler() function like so, which includes some basic logging (for debug purposes) and which accesses certain properties on the `req` object, some of which comes from the original `req` and some of which is added in the `keyGenerator()` function:

    const limitHandler = function (req, res, next, options) {
        const method = req.method;
        const trigger = req.params.options.trigger;
        const source = req.params.options.source;
        const rateLimitTag = req.rateLimitTag;
        const rateLimitKey = req.rateLimitKey;
        console.log(`limitHandler method ${method}`);
        console.log(`limitHandler trigger ${trigger}`);
        console.log(`REJECTING REQUEST (rate limited for ${rateLimitTag} ${rateLimitKey} with source ${source} and trigger ${trigger})`);
        res.header('server-overload', 'true');
        res.sendStatus(429);
    };

    const limiterIP = rateLimit({
        windowMs: 1 * 60 * 1000,
        max: 15,
        standardHeaders: true,
        legacyHeaders: false,
        skipFailedRequests: true,
        skip: function (req, res) {
            return (myTools.getRoute(req) == 'getBasicInfo');
        },
        keyGenerator: function (req, res) {
            const rateLimitKey = MyFunctions.getIP(req);
            req.rateLimitKey = rateLimitKey;
            req.rateLimitTag = 'IP';
            return rateLimitKey;
        },
        handler: limitHandler
    });

What's strange is that I'm seeing the following console ouput:

    REJECTING REQUEST (rate limited for IP aaa.bbb.ccc.ddd with source xyz and trigger zyx)
    REJECTING REQUEST (rate limited for IP ggg.bbb.ccc.rrr with source def and trigger undefined)
    REJECTING REQUEST (rate limited for IP sss.ccc.eee.uuu with source ifj and trigger eds)

in other words... it's only outputting the final one of three `console.log()` calls, and never the first two.  And it's often the case that the `trigger` is `undefined` despite knowing for sure that it is always defined in every incoming request.

I'm probably being stupid...

### Library version

6.4.0

### Node version

16.16.0

### Module system

ESM

**Comment by drmrbrewer** [comment]
Forget it, I figured out what was happening and sure enough it has nothing to do with this library.
